src/rules/importBlacklistRule.ts

```diff
@@ -0,0 +1,106 @@
+import * as Lint from "../index";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "import-blacklist",
+        description: Lint.Utils.dedent`
+            Disallows importing the specified modules directly via \`import\` and \`require\`.
+            Instead only sub modules may be imported from that module.`,
+        rationale: Lint.Utils.dedent`
+            Some libraries allow importing their submodules instead of the entire module.
+            This is good practise as it avoids loading unused modules.`,
+        optionsDescription: "A list of blacklisted modules.",
+        options: {
+            type: "array",
+            items: {
+                type: "string",
+            },
+            minLength: 1,
+        },
+        optionExamples: ["true", '[true, "rxjs", "lodash"]'],
+        type: "typescript",
+        typescriptOnly: false,
+    };
+
+    public static FAILURE_STRING = "This import is not allowed.";
+
+    public isEnabled(): boolean {
+        return super.isEnabled() && this.getOptions().ruleArguments.length > 0;
+    }
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        const options = this.getOptions();
+        return this.applyWithWalker(
+            new NoRequireFullLibraryWalker(sourceFile, options, options.ruleArguments),
+        );
+    }
+}
+
+class NoRequireFullLibraryWalker extends Lint.RuleWalker {
+    constructor (sourceFile: ts.SourceFile, options: Lint.IOptions, private blackList: string[]) {
+        super(sourceFile, options);
+    }
+
+    protected isListed(text: string): boolean {
+        return this.blackList.some((entry) => {
+            return text.substring(1, text.length - 1) === entry;
+        });
+    }
+
+    public visitCallExpression(node: ts.CallExpression) {
+        if (
+            node.expression.getText() === "require" &&
+            node.arguments &&
+            node.arguments[0] &&
+            this.isListed(node.arguments[0].getText())
+        ) {
+            this.failure(node.arguments[0]);
+        }
+        super.visitCallExpression(node);
+    }
+
+    public visitImportEqualsDeclaration(node: ts.ImportEqualsDeclaration) {
+        let moduleReference = <ts.ExternalModuleReference> node.moduleReference;
+        // If it's an import require and not an import alias
+        if (moduleReference.expression) {
+            if (this.isListed(moduleReference.expression.getText())) {
+                this.failure(moduleReference.expression);
+            }
+        }
+        super.visitImportEqualsDeclaration(node);
+    }
+
+    public visitImportDeclaration(node: ts.ImportDeclaration) {
+        if (this.isListed(node.moduleSpecifier.getText())) {
+            this.failure(node.moduleSpecifier);
+        }
+        super.visitImportDeclaration(node);
+    }
+
+    private failure (node: ts.Node): void {
```

Comment:
question: is there any more specific common type for the `node` parameter here that we can declare?

minor: can you please call this method `reportFailure` to be more verbose / readable